src/storage/src/source/types.rs

```diff
@@ -512,49 +458,17 @@ impl SourceReaderMetrics {
     /// Initialises source metrics for a given (source_id, worker_id)
     pub fn new(base: &SourceBaseMetrics, source_id: GlobalId) -> SourceReaderMetrics {
         SourceReaderMetrics {
-            partition_metrics: Default::default(),
             source_id,
             base_metrics: base.clone(),
         }
     }
 
-    /// Log updates to which offsets / timestamps read up to.
-    pub fn metrics_for_partition(&mut self, pid: &PartitionId) -> &SourceReaderPartitionMetrics {
-        self.partition_metrics
-            .entry(pid.clone())
-            .or_insert_with(|| {
-                SourceReaderPartitionMetrics::new(&self.base_metrics, self.source_id, pid)
-            })
-    }
-
     /// Get metrics struct for offset committing.
     pub fn offset_commit_metrics(&self) -> OffsetCommitMetrics {
         OffsetCommitMetrics::new(&self.base_metrics, self.source_id)
     }
 }
 
-/// Partition-specific metrics, recorded to both Prometheus and a system table
-pub struct SourceReaderPartitionMetrics {
```

Comment:
No because the notion of a partition is a source-specific thing and now that we expose progress information in SQL this is also redundant. Our metrics story needs a big overhaul too but one step at a time

Reply:
I agree they need a big overhaul!!